component/src/main/java/org/wso2/extension/siddhi/io/websocket/sink/WebSocketSink.java

```diff
@@ -0,0 +1,196 @@
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "null"
+                ),
+                @Parameter(
+                        name = "headers",
+                        description = "Any specific headers which need to send to the server.\n" +
+                                "The headers should adhere to `'key1:value1', 'key2:value2',...` format.",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "null"
+                ),
+                @Parameter(
+                        name = "idle.timeout",
+                        description = "Idle timeout of the connection",
+                        type = DataType.INT,
+                        optional = true,
+                        defaultValue = "-1"
+                )
+        },
+        examples = {
+                @Example(
+                        syntax = "@Sink(type = ‘websocket’, url = 'ws://localhost:8025/websockets/abc', \n" +
+                                "   @map(type='xml'))\n" +
+                                "define stream Foo (attribute1 string, attribute2 int);",
+                        description = "" +
+                                "A sink of type 'websocket' has been defined.\n" +
+                                "All events arriving at Foo stream via websocket will be sent " +
+                                "to the url ws://localhost:8025/websockets/abc."
+                )
+        }
+)
+
+public class WebSocketSink extends Sink {
+    private StreamDefinition streamDefinition;
+    private String url;
+    private String subProtocol;
+    private String headers;
+    private String idleTimeoutString;
+    private int idleTimeout;
+    private WebSocketClientConnectorListener connectorListener;
+    private HandshakeFuture handshakeFuture = null;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{String.class, ByteBuffer.class};
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[0];
+    }
+
+    @Override
+    protected void init(StreamDefinition streamDefinition, OptionHolder optionHolder,
+                        ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
+        this.streamDefinition = streamDefinition;
+        this.url = optionHolder.validateAndGetStaticValue(WebSocketConstants.URL);
+        this.subProtocol = optionHolder.validateAndGetStaticValue
+                (WebSocketConstants.SUB_PROTOCOL, null);
+        this.headers = optionHolder.validateAndGetStaticValue
+                (WebSocketConstants.HEADERS, null);
+        this.idleTimeoutString = optionHolder.validateAndGetStaticValue
+                (WebSocketConstants.IDLE_TIMEOUT, null);
+        if (idleTimeoutString != null) {
+            idleTimeout = Integer.parseInt(idleTimeoutString);
```

Comment:
This will fail if `idleTimeoutString` is NOT a number-string e.g. `sdasdsad`

Reply:
fixed in 41e1595